opentracing/tracecontext.go

```diff
@@ -43,11 +48,18 @@ type TraceContext interface {
 	// wish to use them as HTTP header keys (or key suffixes), and of course
 	// HTTP headers are case insensitive.
 	//
+	// `restrictedKey` MUST match the regular expression
+	// `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. See
+	// CanonicalizeTraceTagKey().
+	//
```

Comment:
I think the parameter is named `caseInsensitiveKey` not `restrictedKey` ? Also it'd be nice to describe the restriction in english rather/in-addition-to regexp for those of us who are not as literate :)